ecs-agent/api/ecs/client/ecs_client.go

```diff
@@ -710,8 +711,21 @@ func (client *ecsClient) DiscoverServiceConnectEndpoint(containerInstanceArn str
 	return aws.StringValue(resp.ServiceConnectEndpoint), nil
 }
 
-func (client *ecsClient) discoverPollEndpoint(containerInstanceArn string) (*ecsmodel.DiscoverPollEndpointOutput,
+func (client *ecsClient) DiscoverSystemLogsEndpoint(containerInstanceArn string, availabilityZone string) (string,
 	error) {
+	resp, err := client.discoverPollEndpoint(containerInstanceArn, availabilityZone)
+	if err != nil {
+		return "", err
+	}
+	if resp.SystemLogsEndpoint == nil {
```

Comment:
would it be good to check its string value as well? (ie not empty)

Reply:
Not in this case. We care about the nil check as protection against panics but we don't need to check for an empty string because there's no action we can take in that case (we don't want to stop the agent or anything like that.)